src/SharpCompress/Archives/AbstractWritableArchive.cs

```diff
@@ -137,11 +139,11 @@ private bool DoesKeyMatchExisting(string key)
             return false;
         }
 
-        public void SaveTo(Stream stream, WriterOptions options)
+        public async Task SaveToAsync(Stream stream, WriterOptions options)
```

Comment:
CancellationToken parameter should be passed through whole call chain and exposed to library consumers (as non-mandatory parameter on top level).